sentry-core/src/tracing.rs

```diff
@@ -0,0 +1,22 @@
+use std::fmt::{Debug, Formatter};
+use std::sync::Arc;
+
+/// Data passed to the `traces_sampler` function,
+/// which forms the basis for whatever decisions it might make.
+///
+/// TODO: This is a placeholder.
+#[derive(Debug, Default, Clone, Eq, PartialEq)]
+#[non_exhaustive]
+pub struct SamplingContext {}
+
+/// Function to compute tracing sample rate dynamically and filter unwanted traces.
+pub type TraceSampler = Arc<dyn TraceSamplerInner>;
```

Comment:
I would rather explicitly use `Arc<dyn …>` in places where it is used instead of having an `…Inner` trait and type alias.
Ideally in the future, we would have functions for this instead of having people directly write to the props of `ClientOptions`.

Reply:
My understanding is that `Fn` doesn't implement `Debug`, so trait aliasing is necessary. I removed the need for the `Inner` trait but the `TraceSampler` still needs to be there to implement `Debug`.

At least with these changes, the need for `Arc` will be more evident to end-users.